core/server/api/canary/membersStripeConnect.js

```diff
@@ -0,0 +1,18 @@
+const membersService = require('../../services/members');
+
+module.exports = {
+    docName: 'members_stripe_connect',
+    auth: {
+        permissions: true,
+        query(frame) {
+            // This is something you have to do if you want to use the "framework" with access to the raw req/res
+            frame.response = async function (req, res) {
```

Comment:
This is a tough one tbh. This enpoint has completely different concept to all the other ones as it's not "resource-based". Instead, it performs an action - redirect. We only use redirects on middleware level so far and having it here feels very strange. That said, don't have a better concept on where we could put this piece :thinking: 

Does the API has to do the redirect to the `stripeConnectAuthURL` or can the client do the redirect based on the response from the server?

Reply:
> This is a tough one tbh. This enpoint has completely different concept to all the other ones as it's not "resource-based".

Yeah I think our API framework is quite restrictive in that sense, it makes (IMO) valid usecases quite difficult if they deviate from the norm!


> That said, don't have a better concept on where we could put this piece 

Yeah I could think of anywhere better either!

> Does the API has to do the redirect to the stripeConnectAuthURL or can the client do the redirect based on the response from the server?

Technically it could be done on the frontend, a discussion was had, and this was the simpler approach - IMO restructuring so we generate/redirect-to the URL on the client is not worth it though! Would be nicer if controllers could have easier ability to do things like this

Reply:
> Technically it could be done on the frontend, a discussion was had, and this was the simpler approach

After sleeping on it, here is my thinking. If it's not absolutely impairing for the client experience,  we should follow returning "redirect_to_url" property on the returned resource approach. First, tt would be consistent with the resource-oriented convention we try to follow. Second, wont create any hacky solutions which don't fit anywhere in the "frame". 

A good reference for this could be how the redirect URL is generated by Stripe in case of creating a checkout page or [setup intent](https://stripe.com/docs/api/setup_intents/object#setup_intent_object-next_action-redirect_to_url) with `redirect_to_url` property.

I'm probably lacking a lot of context here, so treat this comment purely from a perspective of "how I'd do it the clean way" :sweat_smile: Lmk your thoughts!

Reply:
That solution seems convoluted to me, they'd click a link and we'd make a request, parse the response and then go to some location! That's what the `Location` header is for, and it means being able to use a simple `<a>` tag on the frontend!

I get that you want it to be cleaner, but I really think this needs to be fixed at the "frame" level. We shouldn't be restricted to only supporting certain types of requests because of that, right?

Reply:
Cool. The `<a>` tag usecase makes sense  :+1: Let's allow it as is then and see if we have any similar usecases in the future to modify the "frame"